ci-operator/step-registry/baremetalds/ofcir/setup/baremetalds-ofcir-setup-commands.sh

```diff
@@ -0,0 +1,111 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+set -x
+
+echo "************ baremetalds ofcir setup command ************"
+
+function send_slack(){
+    set +x
+    SLACK_AUTH_TOKEN="T027F3GAJ/B011TAG710V/$(cat $CLUSTER_PROFILE_DIR/slackhook)"
+
+    sleep infinity
+    curl -X POST --data "payload={\"text\":\"<https://prow.ci.openshift.org/view/gs/origin-ci-test/logs/$JOB_NAME/$BUILD_ID|Ofcir setup failed> $1\n\"}" \
+        "https://hooks.slack.com/services/${SLACK_AUTH_TOKEN}"
+    set -x
+}
+
+function exit_with_success(){
+  cat >"${ARTIFACT_DIR}/junit_metal_setup.xml" <<EOF
+  <testsuite name="metal infra" tests="1" failures="0">
+    <testcase name="[sig-metal] should get working host from infra provider"/>
+  </testsuite>
+EOF
+  exit 0
+}
+
+function exit_with_failure(){
+  # TODO: update message to reflect job name/link
+  MESSAGE="baremetalds: ${1:-"Failed to create ci resource: ipi-${NAMESPACE}-${JOB_NAME_HASH}-${BUILD_ID}"}"
+  echo $MESSAGE
+  cat >"${ARTIFACT_DIR}/junit_metal_setup.xml" <<EOF
+  <testsuite name="metal infra" tests="1" failures="1">
+    <testcase name="[sig-metal] should get working host from infra provider">
+      <failure message="">$MESSAGE</failure>
+   </testcase>
+  </testsuite>
+EOF
+  send_slack "$MESSAGE"
+  exit 1
+}
+
+trap 'exit_with_failure' ERR
+
+cd
+
+cat > "${SHARED_DIR}/packet-conf.sh" <<-EOF
+    # Ensure our UID, which is randomly generated, is in /etc/passwd. This is required
+    # to be able to SSH.
+    if ! whoami &> /dev/null; then
+        if [[ -w /etc/passwd ]]; then
+            echo "\${USER_NAME:-default}:x:\$(id -u):0:\${USER_NAME:-default} user:\${HOME}:/sbin/nologin" >> "/etc/passwd"
+        else
+            echo "/etc/passwd is not writeable, and user matching this uid is not found."
+        fi
+    fi
+
+    IP=\$(cat "\${SHARED_DIR}/server-ip")
+    SSHOPTS=(-o 'ConnectTimeout=5' -o 'StrictHostKeyChecking=no' -o 'UserKnownHostsFile=/dev/null' -o 'ServerAliveInterval=90' -o LogLevel=ERROR -i "\${CLUSTER_PROFILE_DIR}/packet-ssh-key")
+
+    # Checkout server
+    for x in \$(seq 10) ; do
+        test "\${x}" -eq 10 && exit 1
+        # Equinix hosts
+        ssh "\${SSHOPTS[@]}" "root@\${IP}" hostname && break
+        # Ironic hosts
+        ssh "\${SSHOPTS[@]}" "centos@\${IP}" sudo dd if=/home/centos/.ssh/authorized_keys of=/root/.ssh/authorized_keys && break
+        sleep 10
+    done
+EOF
+
+function getCIR(){
+    OFCIRURL="https://ofcir-service.ofcir-system.svc.cluster.local/v1/ofcir"
+    OFCIRTOKEN="$(cat ${CLUSTER_PROFILE_DIR}/ofcir-auth-token)"
+    echo "Attempting to acquire a Host from OFCIR"
+    IPFILE=$SHARED_DIR/server-ip
```

Comment:
Is it still needed for backward compatibility? Not sure if it could be removed at all